### PR TITLE
[CST] Update claim letters page to account for appeals

### PR DESCRIPTION
**Pull request by pmclaren19**
## Summary

Added the following behind the feature flag `cst_include_ddl_boa_letters`
- Updated the breadcrumbs of Your Claim Letters page from `Your VA claim letters` to `Your VA claim and appeal letters`
- Updated the header of the Your Claim Letters page from `Your VA claim letters` to `Your VA claim and appeal letters`
- Updated the sub text of the Your Claim Letters page from `You can download your claim letters. We also mail you these letters.` to `You can download your claim letters and appeal decisions. We also mail these to you.`

Updated unit and cypress tests

Updated ClaimLetterListItem.jsx so that the `docTypeToDescription` types are using sentence styling (first letter is capitalized and everything else is lower case)

Changed letter type `Board Of Appeals Decision Letter` to `Board decision`

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#80468

## Testing done

- _Describe what the old behavior was prior to the change_
- _Describe the steps required to verify your changes are working as expected_
- _Describe the tests completed and the results_
- _Exclusively stating 'Specs and automated tests passing' is NOT acceptable as appropriate testing
- _Optionally, provide a link to your [test plan](https://depo-platform-documentation.scrollhelp.site/developer-docs/create-a-test-plan-in-testrail) and [test execution records](https://depo-platform-documentation.scrollhelp.site/developer-docs/execute-tests-in-testrail)_

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Claim Letters Page when  cst_include_ddl_boa_letters is enabled  | Screenshot 2024-04-30 at 11 59 51 AM | Screenshot 2024-05-02 at 2 41 50 PM |
| Claim Letters Page when  cst_include_ddl_boa_letters is disabled  | Screenshot 2024-04-30 at 11 59 51 AM | Screenshot 2024-05-02 at 1 12 19 PM|

## What areas of the site does it impact?

Claim Status Tool

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

### CODE REVIEW

File: src/applications/claims-status/containers/YourClaimLetters/index.jsx [hunk 1]
```diff
@@ -99,7 +112,7 @@ export const YourClaimLetters = ({ isLoading, showClaimLetters }) => {
     );
   };
 
-  if (isLoading) {
+  if (isLoading || isLoadingFeatureFlags) {
```
Comment (jerekshoe) [inline src/applications/claims-status/containers/YourClaimLetters/index.jsx:115]: `isLoading` and `isLoadingFeatureFlags` both point to the same value in the redux store, you can get rid of one of them

**Review by jerekshoe — DISMISSED** [review]

**Comment by skylerschain** [comment]
Just to note in case it wasn't clear from the above: this is behind a feature flag and is not visible to Veterans currently.

**Comment by pmclaren19** [comment]
Talked to @skylerschain and he asked that we update the language to "You can download your claim and appeal letters"

File: src/applications/claims-status/tests/components/claim-letters/ClaimLetterListItem.unit.spec.jsx
Comment (rmessina1010) [inline src/applications/claims-status/tests/components/claim-letters/ClaimLetterListItem.unit.spec.jsx:1]: Just curious about the case insensitive regex usage in this case. Yes the test will pass as currently written,  but  so would 'NoTiFiCaTiOn LeTter'

Comment (pmclaren19) [reply src/applications/claims-status/tests/components/claim-letters/ClaimLetterListItem.unit.spec.jsx:1]: Good point that regex isnt needed ill remove it!

**Comment by pmclaren19** [comment]
Talked to @skylerschain via slack. Added some design changes the subtext will now be behind a feature flag and when the feature flag `cst_include_ddl_boa_letters ` is enabled it will show `You can download your claim letters and appeal decisions. We also mail these to you.`

**Comment by pmclaren19** [comment]
Talked to @skylerschain, he had a meeting with the board and they asked us to change the wording of the `Appeal decision letter` to `Board decision`. Adding change!

**Review by rmessina1010 — DISMISSED** [review]
LGTM

**Review by jerekshoe — DISMISSED** [review]

**Review by jerekshoe — APPROVED** [review]

**Pull request merged**